game/src/core/components/DrawComponent.java

```diff
@@ -340,148 +320,48 @@ public void animationMap(final Map<String, Animation> animationMap) {
-      // this is true for entries like (character/knight/idle_down/idle_down_knight_1.png) and
-      // (character/knight/idle/).
-      if (fileName.startsWith(path.pathString() + "/")) {
-
-        // Get the index of the last FileSeparator; every character after that separator is
-        // part of the filename.
-        int lastSlashIndex = fileName.lastIndexOf("/");
-
-        // Ignore directories, so we only work with strings like
-        // (character/knight/idle_down/idle_down_knight_1.png).
-        if (lastSlashIndex != fileName.length() - 1) {
-          // Get the index of the second-to-last part of the string.
-          // For example, in "character/knight/idle_down/idle_down_knight_1.png", this
-          // would
-          // be the
-          // index of the slash in "/idle".
-
-          int secondLastSlashIndex = fileName.lastIndexOf("/", lastSlashIndex - 1);
-
-          // Get the name of the directory. The directory name is between the
-          // second-to-last and the last separator index.
-          // The directory name serves as the key of the animation in the animation map
-          // (similar to what the IPATh values are for them).
-          // For example: "idle"
-
-          String lastDir = fileName.substring(secondLastSlashIndex + 1, lastSlashIndex);
-
-          // add animation-files to new or existing storage map
-          if (storage.containsKey(lastDir)) storage.get(lastDir).add(new SimpleIPath(fileName));
-          else {
-            LinkedList<IPath> list = new LinkedList<>();
-            list.add(new SimpleIPath(fileName));
-            storage.put(lastDir, list);
+    final Map<String, List<IPath>> subdirectoryMap = new HashMap<>();
+
+    // Walk through the (sub)directories of path2 and fill the subdirectoryMap.
+    Files.walkFileTree(
+        path2,
+        new SimpleFileVisitor<>() {
+          @Override
+          public FileVisitResult visitFile(final Path file, final BasicFileAttributes attrs) {
+            if (Files.isRegularFile(file)) {
+              subdirectoryMap
+                  .computeIfAbsent(
+                      file.getParent().getFileName().toString(), k -> new ArrayList<>())
+                  .add(new SimpleIPath(file.toString()));
+            }
+            return FileVisitResult.CONTINUE;
           }
-        }
-      }
-    }
+        });
 
-    // sort the files in lexicographic order (like the most os)
-    // animations will be played in order
-    storage.values().forEach(x -> x.sort(Comparator.comparing(IPath::pathString)));
-    // create animations
-    storage.forEach(
-        (name, textureSet) -> animationMap.put(name, Animation.fromCollection(textureSet)));
-    jar.close();
+    // A Map with sorted values (IPath lists) in natural string order (ascending)
+    animationMap =
+        subdirectoryMap.entrySet().stream()
+            .collect(Collectors.toMap(Map.Entry::getKey, DrawComponent::getAnimationFromMapEntry));
```

Comment:
Warum machst Du das nicht bereits für jedes einzelne Sub-Dir, was Du bearbeitest? 

Irgendwie ist das alles recht schräg. `Animation` sollte ein Builder-Pattern haben, so dass ich eine Animation per `addPath` schrittweise immer weiter ergänzen kann, und per `build()` kriege ich die fertige `Animation`. Dann kann ich hier einfach für jeden Unterordner eine Animation anlegen und jedes File in dem Unterordner per `addPath` hinzufügen, und wenn ein Unterordner abgearbeitet ist, erzeuge ich die Animation mit `build()` und füge sie der `animationMap` mit dem aktuellen Ordnernamen hinzu.

Reply:
@cagix Also sollte ich hier (in der Klasse ResourceWalker) schon das Builder-Pattern verwenden/einsetzen?

siehe auch https://github.com/Dungeon-CampusMinden/Dungeon/pull/1366#discussion_r1519974436

Reply:
Das war nur laut gedacht. Es gibt #1453 ...

Reply:
> Das war nur laut gedacht. Es gibt #1453 ...

Danke, der Ansicht bin ich (eigentlich) auch ...